pkg/middleware/requestid/requestid.go

```diff
@@ -73,7 +73,25 @@ func (r *handleRequestID) ServerReporter(ctx context.Context, _ interceptors.Cal
 	return interceptors.NoopReporter{}, ctx
 }
 
-func (r *handleRequestID) requestIDFromContext(ctx context.Context) (bool, string, context.Context) {
+func (r *handleRequestID) fromContextOrGenerate(ctx context.Context) (bool, string, context.Context) {
+	haveRequestID, requestID, md := fromContext(ctx)
+
+	if !haveRequestID && r.generateIfMissing {
+		requestID, haveRequestID = r.requestIDGenerator(), true
```

Comment:
why this form?

Reply:
Agreed, we don't use multiple assignments on the same line usually in this codebase.

Reply:
that style was already in place before the PR, but I agree.
Addressed in https://github.com/authzed/spicedb/pull/1845/commits/4e732a92f4ba831c60885c4d27ae5760741ac725